test/rig_tests/proxy/response_from/async_http_test.exs

```diff
@@ -98,4 +98,145 @@ defmodule RigTests.Proxy.ResponseFrom.AsyncHttpTest do
     # ...but the client got the response sent to the HTTP internal endpoint:
     assert Jason.decode!(res_body)["data"] == async_response
   end
+
+  test_with_server "Given response_from is set to http_async and the backend responds with 200, the http response is synchronous." do
+    test_name = "proxy-http-response-synchronous"
+
+    api_id = "mock-#{test_name}-api"
+    endpoint_id = "mock-#{test_name}-endpoint"
+    endpoint_path = "/#{endpoint_id}"
+    sync_response = %{"message" => "this is the sync response that reaches the client"}
+    async_response = %{"this response" => "the client never sees this response"}
+
+    route(endpoint_path, fn %{query: %{"correlation" => correlation_id}} ->
+      event =
+        Jason.encode!(%{
+          specversion: "0.2",
+          type: "rig.async-response",
+          source: "fake-service",
+          id: "1",
+          rig: %{correlation: correlation_id},
+          data: async_response
+        })
+
+      build_conn()
+      |> put_req_header("content-type", "application/json;charset=utf-8")
+      |> post("/v2/responses", event)
+
+      Response.ok!(sync_response, %{"content-type" => "application/json"})
+    end)
+
+    # We register the endpoint with the proxy:
+    rig_api_url = "http://localhost:#{@api_port}/v2/apis"
+    rig_proxy_url = "http://localhost:#{@proxy_port}"
+
+    body =
+      Jason.encode!(%{
+        id: api_id,
+        name: "Mock API",
+        version_data: %{
+          default: %{
+            endpoints: [
+              %{
+                id: endpoint_id,
+                type: "http",
+                method: "GET",
+                path: endpoint_path,
+                response_from: "http_async"
+              }
+            ]
+          }
+        },
+        proxy: %{
+          target_url: "localhost",
+          port: FakeServer.port()
+        }
+      })
+
+    headers = [{"content-type", "application/json"}]
+    HTTPoison.post!(rig_api_url, body, headers)
+
+    # The client calls the proxy endpoint:
+    request_url = rig_proxy_url <> endpoint_path
+    %HTTPoison.Response{status_code: res_status, body: res_body} = HTTPoison.get!(request_url)
+
+    # Now we can assert that...
+    # ...the fake backend service has been called:
+    assert FakeServer.hits() == 1
+    # ...the connection is closed and the status is OK:
+    assert res_status == 200
+    # ...the client receives the synchronous http response:
+    assert Jason.decode!(res_body) == sync_response
+    # ... and never receives the asynchronous response
+    assert Jason.decode!(res_body) != async_response
+  end
+
+  test_with_server "Given response_from is set to http_async and the backend responds with 400, the client should not receive any data." do
```

Comment:
While the implementation is correct, this doesn't sound right - the client indeed receives the 400-response, so it receives data; it just happens that the response has no body. I suggest adding a body like "Bad request from the test endpoint.", which would make the test more readable and also make sure that the response the test sees is actually the response we expect, and not some internal error response we get from the framework.

Keeping with the style above, this could read:

```elixir
  test_with_server "Given response_from=http_async, when the backend responds with 400, RIG forwards this 400-response (and does not expect an asynchronous response for this request)." do
```

Reply:
done